### ISSUE TITLE
Button and TextInput not respecting width

### ISSUE THREAD

**Issue by GCBallesteros** [bokeh/bokeh#8157]
The horizontal layout of a TextInput widget and buttons to its left and right produces overlapping elements when widths are specified. The vertical arrangement is also broken. This issue is probably closely related to #6427 but in this case the Button widget is also not respecting the width.

 Below is the code reproducing the issue.

```
from bokeh.layouts import rowd
from bokeh.models.widgets import TextInput, Button
from bokeh.plotting import  output_file, show

text = TextInput(value="0", width=50)
prev_btn = Button(label="Prev", width=30)
next_btn = Button(label="Next", width=30)

output_file('broken_arrangement.html')
show(row(prev_btn, text, next_btn))
```
Bokeh Version: 0.13.0
Tested on: Firefox 61.0.1, Safari 11.1.1

![screen shot 2018-08-10 at 06 52 21](https://user-images.githubusercontent.com/6859287/43940973-b5e014ea-9c61-11e8-91b9-cee1869bc9bd.png)

I've tested the following combinations of un/setting widths The images below each piece of code correspond to the result.

1) Prev button overlaps text field
```
text = TextInput(value="0")
prev_btn = Button(label="Prev", width=30)
next_btn = Button(label="Next")
```
![screen shot 2018-08-10 at 07 04 43](https://user-images.githubusercontent.com/6859287/43941351-4dfa4a24-9c63-11e8-844e-89edfd3d3c62.png)

2) Text overlaps next button
```
text = TextInput(value="0", width=30)
prev_btn = Button(label="Prev")
next_btn = Button(label="Next")
```
![screen shot 2018-08-10 at 07 06 50](https://user-images.githubusercontent.com/6859287/43941415-982745f2-9c63-11e8-97d2-b487de3905fe.png)

3) Eliminated Text Input. Buttons still overlap if the width of the element to the left is specified.
```
# text = TextInput(value="0")
prev_btn = Button(label="Prev",  width=30)
next_btn = Button(label="Next")
```
![screen shot 2018-08-10 at 07 11 18](https://user-images.githubusercontent.com/6859287/43941578-39ac4ddc-9c64-11e8-8808-6d31b690f0c2.png)

**Comment by bryevdv** [comment]
cc @mattpap any chance this falls out of your layout PR?

**Comment by mattpap** [comment]
Yes. However, one thing to note, is that the existing layout requires all widgets to be wrapped in `WidgetBox` to make them any layoutable. Widgets on their own don't participate in layout in any meaningful way, though can be added to a layout without issue. I changed that and removed `WidgetBox` and made widgets first-class citizens of the layout. So all those widget misalignment issues should go away as soon as the layout PR is merged.

**Comment by mattpap** [comment]
This now work much better and the layout is respected. However, the widget CSS is still lacking, overflowing the layout allocated slots.
